endpoints/openrtb2/auction.go

```diff
@@ -1378,42 +1394,44 @@ func (deps *endpointDeps) processStoredRequests(ctx context.Context, requestJson
 	// Apply any Stored Imps, if they exist. Since the JSON Merge Patch overrides arrays,
 	// and Prebid Server defers to the HTTP Request to resolve conflicts, it's safe to
 	// assume that the request.imp data did not change when applying the Stored BidRequest.
-	impExtInfoMap := make(map[string]exchange.ImpExtInfo, len(impIds))
-	for i := 0; i < len(impIds); i++ {
-		resolvedImp, err := jsonpatch.MergePatch(storedImps[impIds[i]], imps[idIndices[i]])
+	impExtInfoMap := make(map[string]exchange.ImpExtInfo, 0)
+	resolvedImps := make([]json.RawMessage, 0)
```

Comment:
Should `impExtInfoMap` and `resolvedImps` start with a capacity of `len(impInfo)`?

Reply:
Cannot init  `resolvedImps` array with `len(impInfo)` because it fails if some or all impressions in request don't have stored request.

Reply:
Why does it fail? Did you create the `resolvedImps` with a length of 0 and a capacity of `len(impInfo)`? If you just use `make([]json.RawMessage, len(impInfo))` it will create an array with a length `len(impInfo)` with nil jsons for all the entries. You want `make([]json.RawMessage, 0, len(impInfo))` which will create a zero length array, with room preallocated for all the possible entries, so you will never have to reallocate memory as you append to the array.

Reply:
@hhhjort explains it well. Setting the length as the third parameter will preallocate enough memory for the worst case scenario such that Go doesn't need to grow the underlying array. This uses a bit more memory, but saves on cpu and memory by avoiding potential growth logic which would be automatically triggered by using `append`.

Reply:
Oh right, my bad. Fixed.